test/e2e/eventcounter/eventcounter_test.go

```diff
@@ -2099,67 +2291,84 @@ func getFeature(t *testing.T, client featureclient.Client, featureID string) *fe
 	defer cancel()
 	var response *featureproto.GetFeatureResponse
 	var err error
-	numRetries := 6
+	numRetries := 3
 	for i := 0; i < numRetries; i++ {
 		response, err = client.GetFeature(ctx, getReq)
 		if err == nil {
 			break
 		}
 		st, _ := status.FromError(err)
-		if st.Code() == codes.Internal {
-			t.Fatal(err)
+		if st.Code() != codes.Unavailable {
+			return nil, err
 		}
 		fmt.Printf("Failed to get feature. ID: %s. Error code: %d. Retrying in 5 seconds.\n", featureID, st.Code())
 		time.Sleep(5 * time.Second)
 	}
-	return response.Feature
+	return response.Feature, err
+}
+
+func getEvaluationTimeseriesCount(
+	t *testing.T,
+	featureID string,
+	c ecclient.Client,
+	timeRange ecproto.GetEvaluationTimeseriesCountRequest_TimeRange,
+) (*ecproto.GetEvaluationTimeseriesCountResponse, error) {
+	t.Helper()
+	ctx, cancel := context.WithTimeout(context.Background(), timeout)
+	defer cancel()
+	req := &ecproto.GetEvaluationTimeseriesCountRequest{
+		EnvironmentNamespace: *environmentNamespace,
+		FeatureId:            featureID,
+		TimeRange:            timeRange,
+	}
+	var response *ecproto.GetEvaluationTimeseriesCountResponse
+	var err error
+	numRetries := 3
+	for i := 0; i < numRetries; i++ {
+		response, err = c.GetEvaluationTimeseriesCount(
+			ctx,
+			req,
+		)
+		if err == nil {
+			break
+		}
+		st, _ := status.FromError(err)
+		if st.Code() != codes.Unavailable {
```

Comment:
It only retries for all the requests if the server is unavailable due to pods rescheduling, etc.